tools/upgrade-dependencies

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+import json
+import os
+import sys
+import subprocess
+import errno
+
+from typing import List, Dict
+
+ZULIP_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+sys.path.append(ZULIP_PATH)
+
+requirement_files = ["common.in", "dev.in", "docs.in", "prod.in", "thumbor.in"]
+log_dir = "var/log/dependency_upgrade"
+outdated_packages = []  # type: List[Dict[str, str]]
+
+packages_json = subprocess.check_output(["pip", "list", "--outdated", "--format=json"])
+outdated_packages = json.loads(packages_json.decode("utf-8"))
+
+try:
+    os.makedirs(log_dir)
+except OSError as e:
+    if e.errno != errno.EEXIST:
+        raise
+
+for package in outdated_packages:
+    pkg_name = package["name"]
+    version = package["version"]
+    latest_version = package["latest_version"]
+
+    sed_string_template = "s/{name}=={version}/{name}=={latest_version}/g"
+    sed_string = sed_string_template.format(name=pkg_name, version=version, latest_version=latest_version)
+
+    files_updated = False
+    subprocess.check_call(["git", "stash"])
+
+    for requirement_file in requirement_files:
+        file_path = "requirements/{}".format(requirement_file)
+        subprocess.check_call(["sed", "-i", sed_string, file_path])
+
+        diff = subprocess.check_output(["git", "diff", file_path])
+        if diff:
+            subprocess.check_call(["git", "add", file_path])
+            files_updated = True
+
+    if files_updated:
+        try:
+            subprocess.check_output(["./tools/update-locked-requirements"])
+            subprocess.check_output(["./tools/provision"])
+            subprocess.check_output(["./tools/test-all"])
```

Comment:
I think you can just `tools/test-backend`; the other suites are unlikely to fail due to a Python dependency upgrade, and are a lot of time.